buffalo/cmd/fix/middleware.go

```diff
@@ -0,0 +1,157 @@
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+//MiddlewareTransformer moves from our old middleware package to new one
+type MiddlewareTransformer struct {
+	PackagesReplacement map[string]string
+	Aliases             map[string]string
+}
+
+func (mw MiddlewareTransformer) transformPackages(r *Runner) error {
+	return filepath.Walk(".", mw.processFile)
+}
+
+func (mw MiddlewareTransformer) processFile(p string, fi os.FileInfo, err error) error {
+	er := onlyRelevantFiles(p, fi, err, func(p string) error {
+		if err := mw.rewriteMiddlewareUses(p); err != nil {
+			return err
+		}
+
+		// create an empty fileset.
+		fset := token.NewFileSet()
+
+		f, err := parser.ParseFile(fset, p, nil, parser.ParseComments)
+		if err != nil {
+			e := err.Error()
+			msg := "expected 'package', found 'EOF'"
+			if e[len(e)-len(msg):] == msg {
+				return nil
+			}
+			return err
+		}
+
+		//Replacing mw packages
+		for old, new := range mw.PackagesReplacement {
+			deleted := astutil.DeleteImport(fset, f, old)
+			if deleted {
+				astutil.AddNamedImport(fset, f, mw.Aliases[new], new)
+			}
+		}
+
+		if err := mw.addMissingRootMiddlewareImports(fset, f, p); err != nil {
+			return err
+		}
+
+		ast.SortImports(fset, f)
+
+		temp, err := writeTempResult(p, fset, f)
+		if err != nil {
+			return err
+		}
+
+		// rename the .temp to .go
+		return os.Rename(temp, p)
+	})
+
+	return er
+}
+
+func (mw MiddlewareTransformer) addMissingRootMiddlewareImports(fset *token.FileSet, f *ast.File, p string) error {
+	read, err := ioutil.ReadFile(p)
+	if err != nil {
+		return err
+	}
+
+	content := string(read)
+
+	if strings.Contains(content, "paramlogger.ParameterLogger") {
+		astutil.DeleteImport(fset, f, "github.com/gobuffalo/buffalo/middleware")
+		astutil.AddNamedImport(fset, f, "paramlogger", "github.com/gobuffalo/mw-paramlogger")
+	}
+
+	if strings.Contains(content, "popmw.Transaction") {
+		astutil.DeleteImport(fset, f, "github.com/gobuffalo/buffalo/middleware")
+		astutil.AddNamedImport(fset, f, "popmw", "github.com/gobuffalo/buffalo-pop/pop/popmw")
+	}
+
+	if strings.Contains(content, "contenttype.Add") || strings.Contains(content, "contenttype.Set") {
+		astutil.DeleteImport(fset, f, "github.com/gobuffalo/buffalo/middleware")
+		astutil.AddNamedImport(fset, f, "contenttype", "github.com/gobuffalo/mw-contenttype")
+	}
+
+	return ioutil.WriteFile(p, []byte(content), 0)
+}
+
+func (mw MiddlewareTransformer) rewriteMiddlewareUses(p string) error {
+	read, err := ioutil.ReadFile(p)
+	if err != nil {
+		return err
+	}
+
+	newContents := string(read)
+	newContents = strings.Replace(newContents, "middleware.SetContentType", "contenttype.Set", -1)
+	newContents = strings.Replace(newContents, "middleware.AddContentType", "contenttype.Add", -1)
+	newContents = strings.Replace(newContents, "middleware.ParameterLogger", "paramlogger.ParameterLogger", -1)
+	newContents = strings.Replace(newContents, "middleware.PopTransaction", "popmw.Transaction", -1)
+
```

Comment:
it's missing `newContents = string.Replace(newContents, "middleware.ForceSSL", "ssl.Middleware", -1)`

Reply:
Oh, you are right, i only replaced the "root" middleware expressions but may need to also replace the ones using packages as well, i'll also add some tests.

Reply:
Thanks for the review.